### PR TITLE
feat: add dropdown for "folder to create canvas"

### PR DESCRIPTION
**Pull request by qazxcdswe123**
Preview: 
image

___

Also fixed an npm dependency issue, it's used in `lib/canvas/canvas.test.ts`
> (!) Plugin typescript: @rollup/plugin-typescript TS2307: Cannot find module '@jest/globals' or its corresponding type declarations.

### CODE REVIEW

File: lib/suggest/obsidian.d.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// ref: https://github.com/SilentVoid13/Templater/commit/e945c938e5b004c8df12e16c912214e1d5807b6c
```
Comment (benhughes) [inline lib/suggest/obsidian.d.ts:1]: Loaded this up and it's not needed, so can delete

File: lib/suggest/suggest.ts [hunk 2]
```diff
@@ -0,0 +1,193 @@
+// Credits go to Liam's Periodic Notes Plugin: https://github.com/liamcain/obsidian-periodic-notes
```
Comment (benhughes) [inline lib/suggest/suggest.ts:1]: Can you rename this file Suggest so it matches `FolderSuggester`

File: lib/suggest/suggest.ts [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+    scope.register([], 'Enter', (event) => {
+      if (!event.isComposing) {
+        this.useSelectedItem(event);
+        return false;
+      }
+    });
+  }
+
+  onSuggestionClick(event: MouseEvent, el: HTMLDivElement): void {
```
Comment (benhughes) [inline lib/suggest/suggest.ts:54]: if you change this to use arrow functions you won't need to use `.bind` above. Same with the functions below

File: lib/suggest/suggest.ts [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+    this.suggestEl.on(
+      'mousedown',
+      '.suggestion-container',
+      (event: MouseEvent) => {
+        event.preventDefault();
+      }
+    );
+  }
+
+  onInputChanged(): void {
```
Comment (benhughes) [inline lib/suggest/suggest.ts:135]: same with these

File: lib/suggest/suggest.ts [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+    if (suggestions.length > 0) {
+      this.suggest.setSuggestions(suggestions);
+      this.open(app.dom.appContainerEl, this.inputEl);
+    } else {
+      this.close();
+    }
+  }
+
+  open(container: HTMLElement, inputEl: HTMLElement): void {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (benhughes) [inline lib/suggest/suggest.ts:153]: don't think this is needed

File: lib/suggest/FolderSuggester.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
+// Credits go to Liam's Periodic Notes Plugin: https://github.com/liamcain/obsidian-periodic-notes
```
Comment (benhughes) [inline lib/suggest/FolderSuggester.ts:1]: Suprised obsidian doesn't offer this as it's own ui element

File: lib/suggest/suggest.ts [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+    this.owner = owner;
+    this.containerEl = containerEl;
+
+    containerEl.on(
+      'click',
+      '.suggestion-item',
+      this.onSuggestionClick.bind(this)
+    );
+    containerEl.on(
+      'mousemove',
```
Comment (benhughes) [inline lib/suggest/suggest.ts:27]: Why is this needed? Surely a click should set it not a hover?

**Review by benhughes — COMMENTED** [review]
Thanks for this, a nice touch, I know you haven't written the classes that you're using here but would be good to tidy up a few things if you don't mind, so it's consistent with the rest of the codebase.

Can you also run

`npm run pre-commit`

**Comment by qazxcdswe123** [comment]
I totally forgot this and no longer use this plugin, so I guess I shall close this 🥲
